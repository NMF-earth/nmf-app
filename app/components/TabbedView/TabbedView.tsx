import React, { ReactElement, useState } from "react";
import { Button, View } from "react-native";

import styles from "./TabbedView.styles";
import colors from "../../style/colors";

export interface TabbedViewItem {
  title: string;
  component: ReactElement;
}

interface Props {
  items: TabbedViewItem[];
}

interface TabItemProps {
  title: string;
  index: number;
  handleTabPress: (index: number) => void;
  isSelected: boolean;
}

function TabItem({ title, index, handleTabPress, isSelected }: TabItemProps): ReactElement {
  const buttonColor = isSelected ? colors.linkGreen : colors.gray;
  const tabStyles = isSelected ? styles.tabItemSelected : styles.tabItem;

  return (
    <View 
      style={tabStyles}>
      <Button 
        color={buttonColor}
        onPress={() => handleTabPress(index)} 
        title={title} />
    </View>
  );
}

export default function TabbedView(props: Props) {
  const { items } = props;
  const [selectedTab, setSelectedTab] = useState(0);
  

  return (
    <View>
      <View style={styles.tabContainer}>
        {items.map(({ title }, index: number) => (
          <TabItem 
            key={index}
            handleTabPress={setSelectedTab}
            title={title}
            index={index}
            isSelected={index === selectedTab}
          />
      ))}
      </View>
      {items[selectedTab].component}
    </View>
  )
}
