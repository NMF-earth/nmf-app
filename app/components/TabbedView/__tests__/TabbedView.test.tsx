import React from "react";
import renderer from "react-test-renderer";
import TabbedView from "../";
import { Text } from "react-native";

jest.unmock("../");

it("renders correctly TabbedView", () => {
  const items = [
    {
      title: "Hello Tab 1",
      component: <Text>Some details</Text>
    },
    {
      title: "Hello Tab 1",
      component: <Text>Some details</Text>
    }
  ];
  const tree = renderer.create(<TabbedView items={items} />).toJSON();
  expect(tree).toMatchSnapshot();
});
