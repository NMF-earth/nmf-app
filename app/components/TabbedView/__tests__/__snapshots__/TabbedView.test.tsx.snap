// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly TabbedView 1`] = `
<View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#4BB377",
          "borderBottomWidth": 4,
          "flex": 1,
        }
      }
    >
      <TouchableOpacity
        accessibilityRole="button"
        accessibilityStates={Array []}
        onPress={[Function]}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#4BB377",
                },
              ]
            }
          >
            Hello Tab 1
          </Text>
        </View>
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#CED4DA",
          "borderBottomWidth": 4,
          "flex": 1,
        }
      }
    >
      <TouchableOpacity
        accessibilityRole="button"
        accessibilityStates={Array []}
        onPress={[Function]}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#CED4DA",
                },
              ]
            }
          >
            Hello Tab 1
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>
  <Text>
    Some details
  </Text>
</View>
`;
